enclave-runtime/src/top_pool_execution.rs

```diff
@@ -132,21 +130,19 @@ fn execute_top_pool_trusted_calls_internal() -> Result<()> {
 
 	let authority = Ed25519Seal::unseal_from_static_file()?;
 
-	info!("Elapsed time before AURA execution: {} ms", start_time.elapsed().as_millis());
-
 	match yield_next_slot(
 		slot_beginning_timestamp,
 		SLOT_DURATION,
 		latest_parentchain_header,
 		&mut LastSlotSeal,
 	)? {
 		Some(slot) => {
-			let remaining_time = slot.ends_at - slot.timestamp;
-			info!(
-				"Remaining slot time for aura: {} ms, {}% of slot time",
-				remaining_time.as_millis(),
-				(remaining_time.as_millis() as f64 / slot.duration.as_millis() as f64) * 100f64
-			);
+			if slot.duration_remaining().is_none() {
+				warn!("No time remaining in slot, skipping AURA execution");
+				return Ok(())
+			}
+
+			log_remaining_slot_duration(&slot, "Before AURA");
```

Comment:
I added some log messages and timers to help diagnose the timings and phases of the sidechain slot